GraphicsView/include/CGAL/Qt/Basic_viewer_qt.h

```diff
@@ -54,21 +54,41 @@
 
 namespace CGAL
 {
+int& code_to_call_before_creation_of_QCoreApplication(int& i) {
```

Comment:
name of the function: initialize_QCoreApplication ?
I don't understand the reason of i parameter. It is unused in this function, and only return. And why by reference ?

Reply:
The function is to be called as `argc`, it is a trick to call a function before the initialization of a program, so it takes the app's `argc` and returns it, even it doesn't use it as the app will need them.

Reply:
Actually, that function was only *needed* for a special case, involving a constructor, in [`Polyhedron_demo::Polyhedron_demo`](https://github.com/CGAL/cgal/blob/077937383555a61e5b23460fc21cf75f3f005018/Polyhedron/demo/Polyhedron/Polyhedron_demo.cpp#L40-L48).

For any other usage, the initialization just has to be done early in the `main()` function.

Reply:
So, I agree with Guillaune: the name and the usage of it is wrong.

@maxGimeno Create a new function that does the job, and place it in a header that can be used by any of our 3D demo. Then, in `code_to_call_before_creation_of_QCoreApplication` from `Polyhedron/demo/Polyhedron/Polyhedron_demo.cpp`, just call that demo.

And **never** copy-paste, please, when a function can be factorized in a header.

Reply:
And that was needed to keep the API of `Polyhedron_demo` simple, like that:
https://github.com/CGAL/cgal/blob/077937383555a61e5b23460fc21cf75f3f005018/Polyhedron/demo/Polyhedron/Mesh_3.cpp#L11-L18

So, everything had to be done in the constructor, and before the call to the base constructor `QApplication::QApplication`. That is why that function takes a `int&` and returns it untouched: that was a trick to insert the call to `code_to_call_before_creation_of_QCoreApplication` in the arguments of the constructor of `QApplication`.

Reply:
But that trick has useless, and looks strange, in the code of the other 3D demos, so, as I said in https://github.com/CGAL/cgal/pull/5336#discussion_r555579202:
>@maxGimeno Create a new function that does the job, and place it in a header that can be used by any of our 3D demo. Then, in `code_to_call_before_creation_of_QCoreApplication` from `Polyhedron/demo/Polyhedron/Polyhedron_demo.cpp`, just call that demo.